web/template/add_complaint.gohtml

```diff
@@ -51,17 +51,9 @@
 
         {{ template "input" (field "summary" "Title" .Complaint.Summary .Error.Field.summary) }}
 
-        <div class="govuk-form-group {{ if .Error.Field.description }}govuk-form-group--error{{ end }}">
-          <label class="govuk-label" for="f-description">Description</label>
-          {{ template "errors" .Error.Field.description }}
-          <textarea data-module="app-auto-resize" class="govuk-textarea {{ if .Error.Field.description }}govuk-textarea--error{{ end }}" id="f-description" name="description">{{ .Complaint.Description }}</textarea>
-        </div>
+        {{ template "textarea" (field "description" "Description" .Complaint.Description .Error.Field.description) }}
 
-        <div class="govuk-form-group {{ if .Error.Field.receivedDate }}govuk-form-group--error{{ end }}">
-          <label class="govuk-label" for="f-receivedDate">Received date</label>
-          {{ template "errors" .Error.Field.receivedDate }}
-          <input class="govuk-input govuk-!-width-one-third {{ if .Error.Field.receivedDate }}govuk-input--error{{ end }}" type="date" id="f-receivedDate" name="receivedDate" value="{{ .Complaint.ReceivedDate }}" max="{{ today }}" />
-        </div>
+        {{ template "input-date" (field "receivedDate" "Received date" .Complaint.ReceivedDate .Error.Field.receivedDate "max" today) }}
```

Comment:
Not a _massive_ fan of just chaining in the attributes but the only alternative I can think of just makes things more verbose
```suggestion
        {{ template "input-date" (field "receivedDate" "Received date" .Complaint.ReceivedDate .Error.Field.receivedDate "attr" ("max" today)) }}
```

Reply:
and is kind of just chaining attributes itself 😄